model/app.py

```diff
@@ -222,24 +262,28 @@ def key_from_model_info(wild_type_id, message):
     """
     d = {k: message.get(k, []) for k in {GENOTYPE_CHANGES, MEDIUM, MEASUREMENTS, REACTIONS_ADD, REACTIONS_KNOCKOUT}}
     d['model_id'] = wild_type_id
+
+    if version:
+        d['version'] = version
+
     return hashlib.sha224(json.dumps(d, sort_keys=True).encode('utf-8')).hexdigest()
 
 
-async def save_changes_to_db(model, wild_type_id, message):
+async def save_changes_to_db(model, wild_type_id, message, version=None):
     """Store model in cache database
 
     :param model: Cameo model
     :param wild_type_id: str
     :param message: dict
-    :return: str (cache database key)
+    :return: mutated_model_id (cache database key)
     """
-    db_key = key_from_model_info(wild_type_id, message)
+    mutated_model_id = key_from_model_info(wild_type_id, message, version)
     redis = await redis_client()
-    await redis.set(db_key,
+    await redis.set(mutated_model_id,
                     json.dumps({'model': model.id, 'changes': model.notes.get('changes', deepcopy(EMPTY_CHANGES))}))
     redis.close()
-    logger.info('Model created on the base of {} with message {} saved as {}'.format(wild_type_id, message, db_key))
-    return db_key
+    logger.info('Model created on the base of {} with message {} saved as {}'.format(wild_type_id, message, mutated_model_id))
```

Comment:
In general it's better to use old string formatting with logging:
```python
logger.info('Model created on the basis of %s with message %s saved as %s.', wild_type_id, message, mutated_model_id)
```
the difference being that the string is not evaluated/formatted if the logging level is above the info level here.

Generally, my preferred pattern for logging is also to have one logger for each module and make it a global, i.e.,

```python
LOGGER = logging.getLogger(__name__)
```
The logger name can then be made part of the logged message and immediately points to the module where it occurred.